### PR TITLE
Fix various MacOS build warnings

### LINKED ISSUE #467: Various warnings building OQS Provider on Mac
I am hitting various complier warnings building oqs-provider on my Mac.
I have a patch file that addresses these warnings. If worthwhile I can create a PR for this.
 
A sample of them includes....
cmake -S . -B _build && cmake --build _build && ctest --test-dir _build && cmake --install _build

- CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required):

  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

- oqsprov/oqs_sig.c:221:5: warning: initializing 'const unsigned char *' with an expression of type 'char[27]' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]

    "060B6086480186FA6B50080101", // mldsa44_pss2048
[-Wpointer-sign]

- oqsprov/oqs_sig.c:485:62: warning: passing 'char *' to parameter of type 'const uint8_t *' (aka 'const unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]

                if (OQS_SIG_sign(oqs_key, buf, &oqs_sig_len, final_tbs,

- oqsprov/oqs_kem.c:241:18: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]

    if ((evp_ctx == NULL)) {

[oqs_prov_warnings_fix.patch](https://github.com/user-attachments/files/16505610/oqs_prov_warnings_fix.patch)

### PR DESCRIPTION
**Pull request by ashman-p**
Address a few warning see on MacOS M1.
[oqsprovider-MacOS-warnings.txt](https://github.com/user-attachments/files/16557064/oqsprovider-MacOS-warnings.txt)

Fixes #467.

### CODE REVIEW

**Comment by ashman-p** [comment]
I am getting the following error running the formatter manually.
Can anyone give me some pointers to resolve?

@ashman-p ➜ /workspaces/oqs-provider-fix-warnings (main) $ 
@ashman-p ➜ /workspaces/oqs-provider-fix-warnings (main) $   find . -type f -and '(' -name '*.h' -or -name '*.c' -or -name '*.inc' ')' | xargs clang-format --dry-run --Werror
YAML:75:25: error: invalid boolean
AlignConsecutiveMacros: AcrossComments
                        ^~~~~~~~~~~~~~
Error reading /workspaces/oqs-provider-fix-warnings/./.clang-format: Invalid argument
YAML:75:25: error: invalid boolean
AlignConsecutiveMacros: AcrossComments
                        ^~~~~~~~~~~~~~

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 if (WIN32)
 cmake_minimum_required(VERSION 3.20 FATAL_ERROR)
 else()
-cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
```
Comment (thb-sb) [inline CMakeLists.txt:4]: Could you explain why you moved it to `3.5`?

**Review by thb-sb — COMMENTED** [review]
LGTM, except for the upgraded minimum cmake version.

**Comment by thb-sb** [comment]
> I am getting the following error running the formatter manually. Can anyone give me some pointers to resolve?
> 
> @ashman-p ➜ /workspaces/oqs-provider-fix-warnings (main) $ @ashman-p ➜ /workspaces/oqs-provider-fix-warnings (main) $ find . -type f -and '(' -name '_.h' -or -name '_.c' -or -name '*.inc' ')' | xargs clang-format --dry-run --Werror YAML:75:25: error: invalid boolean AlignConsecutiveMacros: AcrossComments ^~~~~~~~~~~~~~ Error reading /workspaces/oqs-provider-fix-warnings/./.clang-format: Invalid argument YAML:75:25: error: invalid boolean AlignConsecutiveMacros: AcrossComments ^~~~~~~~~~~~~~

What version of `clang-format` are you using?

**Comment by baentsch** [comment]
> > I am getting the following error running the formatter manually. Can anyone give me some pointers to resolve?
> > @ashman-p ➜ /workspaces/oqs-provider-fix-warnings (main) $ @ashman-p ➜ /workspaces/oqs-provider-fix-warnings (main) $ find . -type f -and '(' -name '_.h' -or -name '_.c' -or -name '*.inc' ')' | xargs clang-format --dry-run --Werror YAML:75:25: error: invalid boolean AlignConsecutiveMacros: AcrossComments ^~~~~~~~~~~~~~ Error reading /workspaces/oqs-provider-fix-warnings/./.clang-format: Invalid argument YAML:75:25: error: invalid boolean AlignConsecutiveMacros: AcrossComments ^~~~~~~~~~~~~~
> 
> What version of `clang-format` are you using?

This may pertain to https://github.com/open-quantum-safe/oqs-provider/issues/473 (also a `clang` version related formatting issue).

File: CMakeLists.txt (hunk 1 above)
Comment (baentsch) [reply CMakeLists.txt:4]: I wouldn't mind moving it up higher even: 1) `liboqs` [demands 3.15](https://github.com/open-quantum-safe/liboqs/blob/2acfd964b95affd355da080ce9ea9bddd776b708/CMakeLists.txt#L3) and 2) the fewer versions permitted the less support hassle for us.

Comment (thb-sb) [reply CMakeLists.txt:4]: ho I didn't know for `liboqs`, thank you @baentsch and sorry for the noise here…

Comment (ashman-p) [reply CMakeLists.txt:4]: The warning i got indicated < 3.5 will be deprecated ....
"CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required…
…): Compatibility with CMake < 3.5 will be removed from a future version of CMake."

**Comment by baentsch** [comment]
Why did this get closed? The code looked like a valid fix and with the formatting update this should have passed CI, no?

**Comment by ashman-p** [comment]
I was having problems getting coding formatting tests to pass on the branch. In my attempts to address that I sync'ed with your latest your changes in 'main'. That made things worst for the branch so, i decided to start over which closed the PR. Please stay tuned for a new PR.

**Pull request closed without merging**